service/lib/agama/storage/config_solvers/boot.rb

```diff
@@ -65,72 +76,94 @@ def solve_device_alias
           return unless root_device
 
-          partitioned_drive_from_device(root_device)
+          partitionable_from_device(root_device)
         end
 
-        # Recursively looks for the first partitioned drive from the given list of devices.
+        # Recursively looks for the first partitioned config from the given list of devices.
         #
         # @param devices [Array<Configs::Drive, Configs::MdRaid, Configs::VolumeGroup>]
         # @param is_target [Boolean] Whether the devices are target for automatically creating
         #   partitions (e.g., for creating physical volumes).
         #
-        # @return [Configs::Drive, nil]
-        def partitioned_drive_from_devices(devices, is_target: false)
+        # @return [Configs::Drive, Configs::MdDrive, nil]
+        def partitionable_from_devices(devices, is_target: false)
           devices.each do |device|
-            drive = partitioned_drive_from_device(device, is_target: is_target)
+            drive = partitionable_from_device(device, is_target: is_target)
             return drive if drive
           end
 
           nil
         end
 
-        # Recursively looks for the first partitioned drive from the given device.
+        # Recursively looks for the first partitioned config from the given device.
         #
         # @param device [Configs::Drive, Configs::MdRaid, Configs::VolumeGroup]
         # @param is_target [Boolean] See {#partitioned_drive_from_devices}
         #
-        # @return [Configs::Drive, nil]
-        def partitioned_drive_from_device(device, is_target: false)
+        # @return [Configs::Drive, Configs::MdDrive, nil]
+        def partitionable_from_device(device, is_target: false)
           case device
           when Configs::Drive
             (device.partitions? || is_target) ? device : nil
           when Configs::MdRaid
-            partitioned_drive_from_md_raid(device)
+            partitionable_from_md_raid(device)
           when Configs::VolumeGroup
-            partitioned_drive_from_volume_group(device)
+            partitionable_from_volume_group(device)
           end
         end
 
+        # Recursively looks for the first partitioned config from the given MD RAID.
+        #
+        # @param md_raid [Configs::MdRaid]
+        # @return [Configs::Drive, Configs::MdDrive, nil]
+        def partitionable_from_md_raid(md_raid)
+          return partitionable_from_found_md_raid(md_raid) if md_raid.found_device
+
+          partitioned_drive_from_new_md_raid(md_raid)
+        end
+
+        # Recursively looks for the first partitioned config from the given MD RAID.
+        #
+        # @param md_raid [Configs::MdRaid]
+        # @return [Configs::Drive, Configs::MdDrive, nil]
+        def partitionable_from_found_md_raid(md_raid)
+          return md_raid if storage_system.candidate?(md_raid.found_device)
+
+          # TODO: find the correct underlying disk devices for the MD RAID (note they may lack
```

Comment:
Should Agama actually solve this case or simply complain?

Reply:
I think it is doable, so I don't see why we should not try.